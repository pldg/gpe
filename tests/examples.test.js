const candidates = require('./candidates');

const runSnapshotTest = (n) => {
  const example = require(`./examples/example-${n}`);
  const output = example(candidates);

  test(`example-${n} snapshot`, () => expect(output).toMatchSnapshot());
}

describe('Without breakpoints', () => runSnapshotTest(1));
describe('Only defaultBreakpoint', () => runSnapshotTest(2));
describe('One breakpoint', () => runSnapshotTest(3));
describe('consecutiveBreakpoints', () => runSnapshotTest(4));
describe('consecutiveBreakpoints with art', () => runSnapshotTest(5));
describe('One breakpoint and defaultBreakpoint', () => runSnapshotTest(6));

describe('Error art', () => {
  const example = require('./examples/example-7');

  test('example-7', () => expect(() => {
    // toThrow() works only if you call the
    // testing function inside the expect() callback
    // https://stackoverflow.com/a/46707558/
    example(candidates);
  }).toThrow());
});

describe('`extensions`', () => runSnapshotTest(8));
describe('`alt`', () => runSnapshotTest(9));
describe('`className`', () => runSnapshotTest(10));
describe('`publicPath`', () => runSnapshotTest(11));
describe('`src`', () => runSnapshotTest(12));

describe('`src` warning', () => {
  // Place mock function before the execution of the code you're testing
  // https://jestjs.io/docs/en/mock-functions
  console['warn'] = jest.fn();

  const example = require('./examples/example-13');
  const output = example(candidates);

  test('example-13 snapshot', () => {
    expect(output).toMatchSnapshot();
  });

  test('example-13 `src` warning', () => {
    const msg = 'Warning: "600" width doesn\'t exist in candidates, ' +
    'src attribute will fallback to the max width available: "triss_400w.jpg"';

    expect(console.warn).toHaveBeenCalledWith(msg);
  });
});
