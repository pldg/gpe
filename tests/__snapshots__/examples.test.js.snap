// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`alt\` example-9 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    alt=\\"triss\\"
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`className\` example-10 snapshot 1`] = `
"<picture class=\\"triss\\">
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`extensions\` example-8 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    type=\\"image/jpeg\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    src=\\"triss_800w.webp\\"
    type=\\"image/webp\\">
</picture>"
`;

exports[`\`ie9\` example-16 snapshot 1`] = `
"<picture>
  <!--[if IE 9]><video style=\\"display: none\\"><![endif]-->
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <!--[if IE 9]></video><![endif]-->
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`placeholder\` example-15 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    data-src=\\"triss_800w.jpg\\"
    src=\\"data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`prefix\` example-14 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    data-srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    data-srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    data-src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`publicPath\` example-11 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"images/triss_400w.webp 400w, images/triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"images/triss_400w.jpg 400w, images/triss_800w.jpg 800w\\"
    src=\\"images/triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`src\` example-12 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_400w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`\`src\` warning example-13 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`One breakpoint and defaultBreakpoint example-6 snapshot 1`] = `
"<picture>
  <source
    media=\\"(min-width: 768px)\\"
    sizes=\\"50vw\\"
    srcset=\\"triss_close-up_400w.webp 400w, triss_close-up_800w.webp 800w\\"
    type=\\"image/webp\\">
  <source
    media=\\"(min-width: 768px)\\"
    sizes=\\"50vw\\"
    srcset=\\"triss_close-up_400w.jpg 400w, triss_close-up_800w.jpg 800w\\"
    type=\\"image/jpeg\\">
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`One breakpoint example-3 snapshot 1`] = `
"<picture>
  <source
    media=\\"(min-width: 768px)\\"
    sizes=\\"100vw\\"
    srcset=\\"triss_close-up_400w.webp 400w, triss_close-up_800w.webp 800w\\"
    type=\\"image/webp\\">
  <source
    media=\\"(min-width: 768px)\\"
    sizes=\\"100vw\\"
    srcset=\\"triss_close-up_400w.jpg 400w, triss_close-up_800w.jpg 800w\\"
    type=\\"image/jpeg\\">
  <source
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`Only defaultBreakpoint example-2 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"100vw\\"
    srcset=\\"triss_close-up_400w.webp 400w, triss_close-up_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"100vw\\"
    srcset=\\"triss_close-up_400w.jpg 400w, triss_close-up_800w.jpg 800w\\"
    src=\\"triss_close-up_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`Without breakpoints example-1 snapshot 1`] = `
"<picture>
  <source
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`consecutiveBreakpoints example-4 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"(min-width: 768px) 50vw, 100vw\\"
    srcset=\\"triss_400w.webp 400w, triss_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"(min-width: 768px) 50vw, 100vw\\"
    srcset=\\"triss_400w.jpg 400w, triss_800w.jpg 800w\\"
    src=\\"triss_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;

exports[`consecutiveBreakpoints with art example-5 snapshot 1`] = `
"<picture>
  <source
    sizes=\\"(min-width: 768px) 50vw, 100vw\\"
    srcset=\\"triss_close-up_400w.webp 400w, triss_close-up_800w.webp 800w\\"
    type=\\"image/webp\\">
  <img
    sizes=\\"(min-width: 768px) 50vw, 100vw\\"
    srcset=\\"triss_close-up_400w.jpg 400w, triss_close-up_800w.jpg 800w\\"
    src=\\"triss_close-up_800w.jpg\\"
    type=\\"image/jpeg\\">
</picture>"
`;
